pkg/server/events.go

```diff
@@ -104,13 +105,21 @@ func (c *criContainerdService) handleEvent(evt *events.Envelope) {
 			// Non-init process died, ignore the event.
 			return
 		}
-		// Delete the container from containerd.
-		_, err = c.taskService.Delete(context.Background(), &tasks.DeleteTaskRequest{ContainerID: e.ContainerID})
-		// TODO(random-liu): Change isContainerdGRPCNotFoundError to use errdefs.
-		if err != nil && !isContainerdGRPCNotFoundError(err) {
-			// TODO(random-liu): [P0] Enqueue the event and retry.
-			glog.Errorf("Failed to delete container %q: %v", e.ContainerID, err)
-			return
+		task, err := cntr.Container.Task(context.Background(), nil)
+		if err != nil {
+			if !errdefs.IsNotFound(err) {
+				glog.Errorf("failed to stop container, task not found for container %q: %v", e.ContainerID, err)
+				return
+			}
+		}
+		if task != nil {
+			if _, err = task.Delete(context.Background(), containerd.WithProcessKill); err != nil {
```

Comment:
Don't think we want to `KillProcess` here.
If the process is still running, it's a bug.